### ISSUE TITLE
Documentation request: clarify when flags such as `-g` are implicitly added

### ISSUE THREAD

**Issue by jberdine** [ocaml/dune#4075]
I was recently surprised when adding `(ocamlopt_flags ())` to a profile in an `env` stanza resulted in `-g` no longer being passed to `ocamlopt` when building an executable using that profile. This highlighted that I don't know precisely when `dune` implicitly adds such flags and when it doesn't, and I could not find the answer in the documentation. Have I overlooked it somewhere?

For the sake of concreteness, here is the env stanza:
```
(env
 (debug
  (flags
   (-w +a-4-9-18-30-40-42-44-48@50-66 -strict-formats -strict-sequence
     -short-paths -bin-annot -keep-locs -keep-docs -unboxed-types))
  (ocamlopt_flags ())
  (env-vars
   (PPX_TRACE_ENABLED 1))
  (inline_tests enabled)))
```

This is easily worked around by adding `-g` to `flags`, but it would be good to document the expected/current behavior.

**Comment by jeremiedimino** [comment]
Agreed. Here it disappear because `-g` is part of the default value for `ocamlopt_flags` and `(ocamlopt_flags ())` completely overrides the default rather than extend it (as `(ocamlopt_flags (:standard ...))` would).

BTW, you can use the `dune printenv [PATH]` command to see what actual sets of flags Dune will be using in a given directory.
